src/main/java/io/fusionauth/load/FusionAuthBaseWorker.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2012-2025, FusionAuth, All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+package io.fusionauth.load;
+
+import java.util.UUID;
+
+import io.fusionauth.client.FusionAuthClient;
+
+/**
+ * Base worker to create deterministic Ids for FusionAuth load tests.
+ *
+ * @author Brent Halsey
+ */
+public abstract class FusionAuthBaseWorker extends BaseWorker {
+  protected final FusionAuthClient client;
+
+  private final int numberOfApplications;
+
+  private final int numberOfTenants;
+
+  protected UUID applicationId;
+
+  protected int applicationIndex;
+
+  protected UUID tenantId;
+
+  protected int tenantIndex;
+
+  protected FusionAuthClient tenantScopedClient;
+
+  protected int userIndex;
+
+  public FusionAuthBaseWorker(FusionAuthClient client, Configuration configuration) {
+    super(configuration);
+    this.client = client;
+    this.numberOfApplications = configuration.getInteger("numberOfApplications", 0);
+    this.numberOfTenants = configuration.getInteger("numberOfTenants", 0);
+    this.tenantScopedClient = client;
+  }
+
+  public static UUID applicationUUID(int index) {
+    return new UUID(1, index);
+  }
```

Comment:
These two static methods are not referenced outside this class, and I don't see any reason they need to be static based on their usage. If they are always called with the class member values, can we omit the parameter as well?
```suggestion
  private UUID applicationUUID() {
    return new UUID(1, applicationIndex);
  }
```
Or we could set the Ids in these methods. Or we could remove them entirely and just set the class member fields directly from the `setIndex` methods.

It's not wrong, but it bothers me that these are `public static`. I feel like that should be done when it's necessary, but generally methods should have limited visibility.

Reply:
They're simple functions, so I believe keeping them static is more appropriate. Someone might want to add multiple app registrations, so they could call this function with other randomly generated appIds. It doesn't need to be tied to class variables. I don't think limiting visibility is necessary for load testing code that is meant to be experimented with.